framework/src/modules/chain/chain.js

```diff
@@ -188,6 +189,89 @@ module.exports = class Chain {
 		}
 	}
 
+	actions() {
+		return {
+			calculateSupply: action => blockReward.calcSupply(action.params[0]),
+			calculateMilestone: action => blockReward.calcMilestone(action.params[0]),
+			calculateReward: action => blockReward.calcReward(action.params[0]),
+			generateDelegateList: action =>
+				new Promise((resolve, reject) => {
+					this.scope.modules.delegates.generateDelegateList(
+						action.params[0],
+						action.params[1],
+						(err, data) => {
+							if (err) {
+								reject(err);
+							}
+
+							resolve(data);
+						},
+						action.params[2]
+					);
+				}),
+			getNetworkHeight: async action =>
+				util.promisify(this.scope.modules.peers.networkHeight)(
+					action.params[0]
+				),
+			getTransactionsCount: async () =>
+				util.promisify(
+					this.scope.modules.transactions.shared.getTransactionsCount
+				)(),
+			updateForgingStastus: async action =>
+				util.promisify(this.scope.modules.delegates.updateForgingStatus)(
+					action.params[0],
+					action.params[1],
+					action.params[2]
+				),
+			getPeers: async action =>
+				this.scope.modules.peers.shared.getPeers(
+					action.params[0],
+					action.params[1]
+				),
+			getPeersCountByFilter: async action =>
+				this.scope.modules.peers.shared.getPeersCountByFilter(action.params[0]),
+			postSignature: async action =>
+				this.scope.modules.signatures.shared.postSignature(
+					action.params[0],
+					action.params[1]
+				),
+			storageRead: async action =>
+				this.scope.logic.block.storageRead(action.params[0]),
+			getLastConsensus: async () => this.scope.modules.peers.getLastConsensus(),
+			loaderLoaded: async () => this.scope.modules.loader.loaded(),
+			loaderSyncing: async () => this.scope.modules.loader.syncing(),
```

Comment:
Other possibility could be to have one action `getChainStatus` that send you these information `loaded` or `syncing` etc. This will simplify, I just want to have least actions possible here. But do it in separate issue :)

Reply:
I agree with you, will open a separate issue to solve all of this :)

Reply:
Issue: https://github.com/LiskHQ/lisk/issues/2993